controllers/csi/metadata/correctness.go

```diff
@@ -0,0 +1,67 @@
+package metadata
+
+import (
+	"context"
+
+	dynatracev1alpha1 "github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Checks if the entries in the storage are actually valid
+// Removes not valid entries
+func CheckStorageCorrectness(cl client.Client, access Access, log logr.Logger) error {
```

Comment:
If it not only checks the storage but also removes invalid entries, it should be called something like `CorrectStorage`